Realm/Tests/RealmTests.mm

```diff
@@ -175,6 +177,58 @@ - (void)testFileFormatUpgradeRequiredButDisabled {
     RLMAssertThrowsWithCodeMatching([RLMRealm realmWithConfiguration:config error:nil], RLMErrorFileFormatUpgradeRequired);
 }
 
+- (void)testExceedingVirtualAddressSpace {
+    RLMRealmConfiguration *config = [RLMRealmConfiguration defaultConfiguration];
+
+    const NSUInteger stringLength = 1024 * 1024;
+    void *mem = calloc(stringLength, '1');
+    NSString *largeString = [[NSString alloc] initWithBytesNoCopy:mem
+                                                           length:stringLength
+                                                         encoding:NSUTF8StringEncoding
+                                                     freeWhenDone:YES];
+
+    @autoreleasepool {
+        RLMRealm *realm = [RLMRealm realmWithConfiguration:config error:nil];
+        [realm beginWriteTransaction];
+        StringObject *stringObj = [StringObject new];
+        stringObj.stringCol = largeString;
+        [realm addObject:stringObj];
+        [realm commitWriteTransaction];
+    }
+
+    struct VirtualMemoryChunk {
+        vm_address_t address;
+        vm_size_t size;
+    };
+    typedef struct VirtualMemoryChunk VirtualMemoryChunk;
```

Comment:
This `typedef` shouldn't be necessary since this code is in an Objective-C++ file.